libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1337,6 +1337,20 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context << Instruction::DUP2 << Instruction::ISZERO;
 		m_context.appendConditionalJumpTo(m_context.errorTag());
 
+		// Test for (minInt) / (-1)
+		if (c_isSigned && _operator == Token::Div)
+		{
+			m_context << Instruction::DUP1 // x
+				<< u256(255) << u256(2) << Instruction::EXP // 2 ** 255
+				<< Instruction::EQ;
+			// y, x, (x == 2 ** 255)
+			m_context << Instruction::DUP3 << Instruction::NOT //y
+				<< Instruction::ISZERO;
+			// y, x, (x == 2 ** 255), (y == -1)
+			m_context << Instruction::AND;
+			m_context.appendConditionalJumpTo(m_context.errorTag());
+		}
+
 		if (_operator == Token::Div)
 			m_context << (c_isSigned ? Instruction::SDIV : Instruction::DIV);
 		else
```

Comment:
The `SMOD` specification is [ambiguous](https://github.com/ethereum/yellowpaper/issues/190).  I created [a separate issue](https://github.com/ethereum/solidity/issues/1162) about this.